src/main/java/jp/co/yahoo/yosegi/tools/FindBestSortKey.java

```diff
@@ -0,0 +1,333 @@
+      if( skipCount < 10000 ) {
+        skipCount++;
+        continue;
+      }
+      writer.write( reader.next() );
+      inCount++;
+    }
+    writer.close();
+    reader.close();
+  }
+
+  public static List<KeyAndRealDataSize> getColumnStats( final String input ) throws IOException {
+    Configuration config = new Configuration();
+
+    InputStream in = FileUtil.fopen( input );
+    YosegiStatsReader reader = new YosegiStatsReader();
+    File file = new File( input );
+    long fileLength = file.length();
+    reader.readStream( in , fileLength , config );
+    ColumnStats stats = reader.getTotalColumn();
+    List<KeyAndRealDataSize> list = new ArrayList<KeyAndRealDataSize>();
+    List<String> currentKeyList = new ArrayList<String>();
+    addChildColumnStats( currentKeyList , list , stats.getChildColumnStats() );
+    Collections.sort( list , Collections.reverseOrder() );
+    return list;
+  }
+
+  public static void addChildColumnStats(
+      final List<String> parentKeyList,
+      final List<KeyAndRealDataSize> list ,
+      final Map<String,ColumnStats> childColumns ) {
+    for ( Map.Entry<String,ColumnStats> entry : childColumns.entrySet() ) {
+      if ( entry.getKey().equals( "ARRAY" ) ) {
+        continue;
+      }
+      List<String> currentKeys = new ArrayList<String>( parentKeyList );
+      currentKeys.add( entry.getKey() );
+      long dataSize = entry.getValue().getTotalStats().getRealDataSize();
+      list.add( new KeyAndRealDataSize( currentKeys , Long.valueOf( dataSize ) ) );
+      addChildColumnStats( currentKeys , list , entry.getValue().getChildColumnStats() );
+    }
+  }
+
+  public static void createColumsFile(
+      final String outputPath , final List<KeyAndRealDataSize> list ) throws IOException {
+    File file = new File( outputPath );
+    BufferedWriter bw = new BufferedWriter( new FileWriter( file ) );
+    for( KeyAndRealDataSize k : list ) {
+      bw.write( k.toString() );
+      bw.newLine();
+    }
+    bw.close();
+  }
+
+  public static KeyAndRealDataSize runCreateFile(
+      final String target , final String tmpPath , final KeyAndRealDataSize k ) throws IOException {
+    Process process = new ProcessBuilder ( "/bin/sort" , "-S128m" ).start();
+    OutputStream sortOut = process.getOutputStream();
+    InputStream in = FileUtil.fopen( target );
+    IStreamReader reader = StreamReaderFactory.create( in , "json" , null );
+    JacksonMessageWriter jsonWriter = new JacksonMessageWriter();
+    while ( reader.hasNext() ) {
+      IParser parser = reader.next();
+      IParser currentParser = parser;
+      for( int i = 0 ; i < k.keys.size() - 1 ; i++ ) {
+        currentParser = currentParser.getParser( k.keys.get( i ) );
+      }
+      byte[] sortKey = null;
+      String sortKeyStr = currentParser.get( k.keys.get( k.keys.size() - 1 ) ).getString();
+      if ( sortKeyStr != null ) {
+        sortKey = sortKeyStr.replace( "" , "" ).getBytes();
```

Comment:
Could you please teach me to use following expression?
It seems useless. (however, it might have some function that I don't know...)
replace( "" , "" )

If this expression has some effect (and need it), please describe as comment to help code reading.